### ISSUE TITLE
Unhelpful memo with fints

### ISSUE THREAD

**Issue by therealppa** [schurig/ynab-bank-importer#31]
With my Sparkasse account, the memo field get filled with (to me) unrecognizable data like:
EREF+123456789012 PP.1234.

The actual memo (that e.g. MoneyMoney shows) in that case was:
PP.1234PP . Cinemaxx, Ihr Einkauf bei Cinemaxx

Which is a lot more helpful. Any chance to have the second value instead?

**Comment by schurig** [comment]
The issue here lays on the FinTS-Protocol side I think. But I know that other apps do manage it to somehow only display the relevant information. I'll try to find some time in the upcoming weeks to have a look how I can remove the not-useful information and only take the "Verwendungszweck".

As written over here https://github.com/schurig/ynab-bank-importer/issues/30#issuecomment-429567127 that would also help me to debug what exactly your bank returns, so that I can check if it's the same for most banks, this way I can be sure not to break other banks. 😊

**Comment by f2cx** [comment]
I had the similiar issue using **DKB** accounts. As quick workaround I just extract the field ``sepa["SVWZ"]`` (if existent) from the transaction object instead of ``sub_fields["20"]``.

JSON Data of an Example Paypal Transaction Object **(DKB account)**:

```
{
  "sha": "XXXXXXXX",
  "date": "2018-09-27",
  "entry_date": "2018-09-27",
  "amount": 167.81,
  "amount_in_cents": 16781,
  "sign": -1,
  "debit": true,
  "credit": false,
  "funds_code": "D",
  "swift_code": "NDDT",
  "reference": "NONREF",
  "bank_reference": "",
  "currency_letter": "R",
  "bic": "XXXXXXXX",
  "iban": "XXXXXXXX",
  "name": "PayPal (Europe) S.a.r.l. et  Cie., S.C.A.",
  "sepa": {
    "EREF": "XXXXXXXX PP.XXXX.PP PAYPAL",
    "MREF": "XXXXXXXX",
    "CRED": "XXXXXXXX",
    "SVWZ": "PP.XXXX.PP . XXXXXXXX, Ihr Einkauf bei XXXXXXXX"
  },
  "information": "EREF+XXXXXXXX PP.XXXX.PP PAYPALMREF+XXXXXXXX+XXXXXXXX+PP.XXXX.PP . XXXXXXXX, Ihr Einkauf bei XXXXXXXX",
  "description": "FOLGELASTSCHRIFT",
  "sub_fields": {
    "00": "FOLGELASTSCHRIFT",
    "10": "XXXX",
    "20": "EREF+XXXXXXXX PP.XXX.",
    "21": "PP PAYPAL",
    "22": "MREF+XXXXXXXX",
    "23": "CRED+XXXXXXXX",
    "24": "XXXX",
    "25": "SVWZ+PP.XXXX.PP . XXXXXXXX",
    "26": "FA, Ihr Einkauf bei XXXXXXXX",
    "27": "XXXX",
    "30": "XXXXXXXX",
    "31": "XXXXXXXX",
    "32": "PayPal (Europe) S.a.r.l. et",
    "33": " Cie., S.C.A.",
    "34": "XXX"
  },
  "transaction_code": "105",
  "details": "?00FOLGELASTSCHRIFT?XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
}

```
Would be interesting to see the transaction object from a **Sparkasse account** to compare the fields...

**Comment by schurig** [comment]
Hi @f2cx thanks for sharing this info. Where did you get the json from? I couldn't find most of those attributes in my logs. The data it returns in `sepa` seems really nice!

**Comment by f2cx** [comment]
For debug purposes, I included a debug statement here:

https://github.com/f2cx/ynab-bank-importer/blob/7161a8c187dc0367dc8b60630f335b9b13546849/lib/dumper/fints.rb#L78

I do not have any other FinTS account to compare the output with, so the ``sepa`` fields might be specific for [DKB](https://www.dkb.de/) accounts. Seems to be quite nice and convenient, because no parsing of the weird sub_fields is needed...

**Comment by BluetriX** [comment]
Same with Sparda-Bank Hessen:

{
  "sha": "xxxxx",
  "date": "2019-01-07",
  "entry_date": "2019-01-07",
  "amount": 11.87,
  "amount_in_cents": 1187,
  "sign": -1,
  "debit": true,
  "credit": false,
  "funds_code": "D",
  "swift_code": "NMSC",
  "reference": "NONREF",
  "bank_reference": "",
  "currency_letter": null,
  "bic": null,
  "iban": null,
  "name": "AMAZON EU S.A R.L., NIEDERL",
  "sepa": {
    "EREF": " 1343949XXX128524 ",
    "MREF": " BV5ocyd,XXXXXX-q5iW)bj6X+Pg ",
    "CRED": " DE24ZZZXXXX652 ",
    "SVWZ": " 302-55XXXXX6-1257111 Amazon.de 134394XXXXX28524"
  },
  "information": "ASSUNG DEUTSCHLANDSEPA-BASISLASTSCHRIFTEREF+ 1343949677128524 MREF+ BV5ocyd,+.JXXXvPK-q5iW)bj6X+Pg CRED+ DE24ZZZ00000561652 SVWZ+ 302-5558606-1257111 Amazon.de 1343949677128524",
  "description": "?20ASSUNG DEUTSCHLAND?21SEPA-BASISLASTSCHRIFT?22EREF+ 1343949677128524 MREF?23+ BV5ocyd,+.JXXXvPK-q5iW)bj?246X+Pg CRED+ DE24ZZZXXXXXX61?25652 SVWZ+ 302-5558606-12571?2611 Amazon.de 1343949XXX1285?2724?32AMAZON EU S.A R.L., NIEDERL",
  "sub_fields": {
    "20": "ASSUNG DEUTSCHLAND",
    "21": "SEPA-BASISLASTSCHRIFT",
    "22": "EREF+ 1343949XXXXX8524 MREF",
    "23": "+ XXXX,+.XXXX-q5iW)bj",
    "24": "6X+Pg CRED+ DE24XXXXXX00561",
    "25": "652 SVWZ+ 302-5558606-12571",
    "26": "11 Amazon.de 1343949XXX1285",
    "27": "24",
    "32": "AMAZON EU S.A R.L., NIEDERL"
  },
  "transaction_code": "835",
  "details": "?20ASSUNG DEUTSCHLAND?21SEPA-BASISLASTSCHRIFT?22EREF+ 1343949XXX128524 MREF?23+ BV5ocyd,+.JXXXvPK-XXXXXj?246X+Pg CRED+ DE24ZZZXXXXXX61?25652 SVWZ+ 302-5XX8606-12571?2611 Amazon.de 1343949XXX1285?2724?32AMAZON EU S.A R.L., NIEDERL"
}


I think as memo SVWZ would be great. And is there a reason that the Payee is shortened?

**Comment by 117agu** [comment]
> For debug purposes, I included a debug statement here:
> 
> https://github.com/f2cx/ynab-bank-importer/blob/7161a8c187dc0367dc8b60630f335b9b13546849/lib/dumper/fints.rb#L78
> 
> I do not have any other FinTS account to compare the output with, so the `sepa` fields might be specific for [DKB](https://www.dkb.de/) accounts. Seems to be quite nice and convenient, because no parsing of the weird sub_fields is needed...

I've tried this by adding put JSON  at the same location as in your file but I don't know where to find the JSON afterwards...

**Comment by schurig** [comment]
I just merged #53. Hopefully this solves the problem.

Please be aware that using the latest version might result into importing duplicates because the `import_id` changed. Right after the import you will see them in YNAB and can select them and delete them. This way they won't be imported again.

This will only effect transactions that are younger than 35 days.

**Comment by kai-truempler** [comment]
Hi, with my DKB account, I'm still running into a similar problem.
The ynab memo field gets: 

> FOLGELASTSCHRIFT / EREF+Z30PAQ1MSUFCMREF+MS0000002DYXKS2001XXXX+GB22357SDDCITI00000004101353SVWZ+Mic

while the memo on the web interface is:

> Microsoft.Office 365 Home AWV-MELDEPFLICHT BEACHTEN HOTLINE BUNDESBANK: (0800) 1234-111

Is there something I can do?
